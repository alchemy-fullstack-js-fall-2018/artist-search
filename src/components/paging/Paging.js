import React from 'react';
import styles from './Paging.css';
import PropTypes from 'prop-types';

const Paging = ({ currentPage, totalPages, updatePage }) => {
  let previousButton = null;
  if(currentPage > 1) {
    previousButton = (
      <button onClick={() => updatePage(currentPage - 1)}>&lt;</button>
    );
  }
  let nextButton = null;
  if(currentPage < totalPages) {
    nextButton = (
      <button onClick={() => updatePage(currentPage + 1)}>&gt;</button>
    );
  }
  return (
    <div className={styles.paging}>
      {previousButton}
      <span>
        {currentPage} out of {totalPages}
      </span>
      {nextButton}
    </div>
  );
};

Paging.propTypes = {
  currentPage: PropTypes.number,
  totalPages: PropTypes.number,
  updatePage: PropTypes.func
};

export default Paging;
