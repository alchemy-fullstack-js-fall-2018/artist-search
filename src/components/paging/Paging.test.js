import React from 'react';
import Paging from './Paging';
import { shallow } from 'enzyme';

describe('Paging', () => {
  it('renders the Paging component', () => {
    const wrapper = shallow(<Paging currentPage={2} totalPages={20} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('decrements the currentPage when previous is clicked', () => {
    const updatePageFn = jest.fn();
    const wrapper = shallow(
      <Paging currentPage={10} totalPages={10} updatePage={updatePageFn} />
    );

    wrapper
      .find('button')
      .at(0)
      .simulate('click');

    expect(updatePageFn.mock.calls[0][0]).toEqual(9);
  });

  it('decrements the currentPage when previous is clicked', () => {
    const updatePageFn = jest.fn();
    const wrapper = shallow(
      <Paging currentPage={10} totalPages={20} updatePage={updatePageFn} />
    );

    wrapper
      .find('button')
      .at(1)
      .simulate('click');

    expect(updatePageFn.mock.calls[0][0]).toEqual(11);
  });

  it('the previous button is not displayed on the first page', () => {
    const wrapper = shallow(
      <Paging currentPage={1} totalPages={20} updatePage={() => {}} />
    );

    expect(wrapper.find('button')).toHaveLength(1);
    expect(
      wrapper
        .find('button')
        .at(0)
        .text()
    ).toEqual('>');
  });

  it('the next button is not displayed on the last page', () => {
    const wrapper = shallow(
      <Paging currentPage={20} totalPages={20} updatePage={() => {}} />
    );

    expect(wrapper.find('button')).toHaveLength(1);
    expect(
      wrapper
        .find('button')
        .at(0)
        .text()
    ).toEqual('<');
  });
});
