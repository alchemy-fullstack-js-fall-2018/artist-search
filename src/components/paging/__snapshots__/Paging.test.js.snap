// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging renders the Paging component 1`] = `
<div
  className="paging"
>
  <button
    onClick={[Function]}
  >
    &lt;
  </button>
  <span>
    2
     out of 
    20
  </span>
  <button
    onClick={[Function]}
  >
    &gt;
  </button>
</div>
`;
